### PR TITLE
rec: Prevent duplicate C/DNAMEs being included when doing serve-stale

### PR DESCRIPTION
**Pull request by omoerbeek**
This can happen if the CNAME record itself was found, but its target not

Reported by @phonedph1  on IRC.

### Checklist

I have:
- [X] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [X] compiled this code
- [X] tested this code
- [ ] included documentation (including possible behaviour changes)
- [X] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)

Needs test

### CODE REVIEW

File: pdns/recursordist/syncres.cc [hunk 1]
```diff
@@ -1733,7 +1733,7 @@ int SyncRes::doResolve(const DNSName& qname, const QType qtype, vector<DNSRecord
       if (child == qname) {
         LOG(prefix << qname << ": Step3 Going to do final resolve" << endl);
         res = doResolveNoQNameMinimization(qname, qtype, ret, depth, beenthere, context);
-        LOG(prefix << qname << ": Step3 Final resolve: " << RCode::to_s(res) << "/" << ret.size() << endl);
+        LOG(prefix << qname << ": Step3 Final resolve: " << (res >= 0 ? RCode::to_s(res) : std::to_string(res)) << "/" << ret.size() << endl);
```
Comment (Habbie) [inline pdns/recursordist/syncres.cc:1736]: why this change?

**Review by Habbie — APPROVED** [review]
approved with one unimportant question

File: pdns/recursordist/syncres.cc (hunk 1 above)
Comment (omoerbeek) [reply pdns/recursordist/syncres.cc:1736]: SyncRes overloads the return value, it can be an rcode or negative for special cases.
RCode::to_s(res) prints `"ErrOutOfRange"` in that case.

Comment (Habbie) [reply pdns/recursordist/syncres.cc:1736]: ok!
